packages/esbuild-plugin-node/README.md

```diff
@@ -0,0 +1,189 @@
+const { build } = require('esbuild');
+
+build({
+  entryPoints: ['src/server.ts'],
+  bundle: true,
+  outfile: 'dist/server.js',
+  target: 'node20',
+  platform: 'node',
+  sourcemap: true,
+  plugins: [openTelemetryPlugin()],
+});
+```
+
+## Usage: Instrumentation Initialization
+
+OpenTelemetry Meta Packages for Node automatically loads instrumentations for Node builtin modules and common packages.
+
+Enable auto instrumentation by configuring it in your esbuild script:
+
+```javascript
+const { openTelemetryPlugin } = require('@opentelemetry/esbuild-plugin-node');
+const { build } = require('esbuild');
+
+build({
+  entryPoints: ['src/server.ts'],
+  bundle: true,
+  outfile: 'dist/server.js',
+  target: 'node20',
+  platform: 'node',
+  sourcemap: true,
+  plugins: [openTelemetryPlugin()],
+});
+```
+
+Custom configuration for each of the instrumentations can be passed to the plugin, by providing an object with the name of the instrumentation as a key, and its configuration as the value.
+
+```javascript
+const { openTelemetryPlugin } = require('@opentelemetry/esbuild-plugin-node');
+const { build } = require('esbuild');
+
+build({
+  entryPoints: ['src/server.ts'],
+  bundle: true,
+  outfile: 'dist/server.js',
+  target: 'node20',
+  platform: 'node',
+  sourcemap: true,
+  plugins: [
+    openTelemetryPlugin({
+      instrumentationConfig: {
+        '@opentelemetry/instrumentation-aws-sdk': {
+          suppressInternalInstrumentation: true,
+        },
+      },
+    }),
+  ],
+});
+```
+
+This esbuild script will instrument non-builtin packages but will not configure the rest of the OpenTelemetry SDK to export traces
+from your application. To do that you must also configure the SDK.
+
+The esbuild script currently only patches non-builtin modules (more specifically, modules in [opentelemetry-js-contrib](https://github.com/open-telemetry/opentelemetry-js-contrib)), so this is also the place to configure the instrumentation
+for builtins or add any additional instrumentations.
+
+### Gotchas
+
+There are limitations to the configuration options for each package. Most notably, any functions (like `ignoreIncomingRequestHook` in the example) are not allowed to be passed in to plugins.
+
+The reason for this is that the current mechanism of instrumenting packages involves stringifying the instrumentation configs, which does not account for any external scoped dependencies, and thus creates subtle opportunities for bugs.
+
+```javascript
+const {
+  getNodeAutoInstrumentations,
+} = require('@opentelemetry/auto-instrumentations-node');
+const {
+  AsyncHooksContextManager,
+} = require('@opentelemetry/context-async-hooks');
+const {
+  OTLPTraceExporter,
+} = require('@opentelemetry/exporter-trace-otlp-http');
+const { NodeSDK } = require('@opentelemetry/sdk-node');
+
+const instrumentations = getNodeAutoInstrumentations({
+  '@opentelemetry/instrumentation-http': {
+    ignoreIncomingRequestHook: request => {
```

Comment:
This README needs some updating, like removing functions from config args, but I'm planning on holding off until the code is more agreed upon to avoid toil in updating the README